### ISSUE TITLE
AsyncStorage setItem Crashed on iOS

### ISSUE THREAD

**Issue by Karanp13** [react-native-community/react-native-async-storage#110]
## Current behavior
AsyncStorage setItem Crashed on iOS

## Expected behavior
Value Set but application crashed.

## Initialise Values
constructor(props) {
    super(props);
     this.state={
      valueOneResult:1,
    }
     this._isMounted = false; 
  }
   componentDidMount() {
     this._isMounted = true;
  }
  componentWillUnmount(){
     this._isMounted = false;
## Repro steps
FinalUpdates = async ()=>{
    try{
    await AsyncStorage.setItem('volumeUpValue',this.state.valueOneResult);
    } catch (error) {
      console.log('error', error);
    }
  };
## Update Value 
this._isMounted && this.setState({valueOneResult : 10})
## Function Call 
<TouchableOpacity
          onPress={() => { 
               this.FinalUpdates(),this.props.navigation.navigate('VolumeDown')
          }} >  
          <Text>Click Me</Text>
           </TouchableOpacity> 
        </View>

## Environment
React Native Environment Info:
    System:
      OS: macOS 10.14.4
      CPU: (2) x64 Intel(R) Pentium(R) CPU 2127U @ 1.90GHz
      Memory: 147.42 MB / 8.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.14.1 - /usr/local/bin/node
      npm: 6.9.0 - /usr/local/bin/npm
    SDKs:
      iOS SDK:
        Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
    IDEs:
      Xcode: 10.2/10E125 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.3 => 16.8.3 
      react-native: ^0.59.8 => 0.59.8 
    npmGlobalPackages:
      create-react-native-app: 2.0.2
      react-native-cli: 2.0.1
- **Async Storage version**: 
"@react-native-community/async-storage": "^1.3.4"
- **React-Native version**: 
"react": "16.8.3",
    "react-native": "^0.59.8",
- **Platform tested**: 
iOS
- **Logs/Error that are relevant**:

**Comment by Krizzu** [comment]
Hey @Karanp13 ,

Can you provide stack trace of that error?


thanks.

**Comment by Karanp13** [comment]
Actually locally not shows any error, when I live that application and shows me crashed. 
Can you please check the setitem values?
'await AsyncStorage.setItem('volumeUpValue',this.state.valueOneResult);'

**Comment by tido64** [comment]
@Karanp13 We are going to need logs and stack traces. The best would be if you can create a minimal repro. 😄

**Comment by Karanp13** [comment]
I solved using JSON.parse and JSON.stringify, Thanks

**Comment by Foveluy** [comment]
nope, `iOS` crash still happend when 
```
let value = 1;
type of value === 'number' //true
AsyncStorage.setItem(KEY,value);
```
the code above will crash on release mode, but on debug mode, it is fine.

to fix this, you should do things like below 

```
let value = "1";
type of value === 'number' //false
AsyncStorage.setItem(KEY,value);

```

**Comment by dholbrook86** [comment]
This was a nightmare! If there is a type error please let me know while I'm developing and not when I push out to TestFlight! Come on React...

**Comment by tido64** [comment]
@dholbrook86 For what it's worth, we addressed this in #231.

**Comment by dholbrook86** [comment]
@tido64 Fair enough. I’m on an older version. Going to update soon.
